esx_service/vmdk_ops.py

```diff
@@ -1038,6 +1040,7 @@ def disk_detach_int(vmdk_path, vm, device):
     spec.deviceChange = dev_changes
 
     try:
+        si = get_si()
```

Comment:
Is there a reason to call get_si() from within the try-except block, in the test and utils code its not called that way. Not wrong just asking.